### PR TITLE
Adds fields and template for course landing page

### PR DESCRIPTION
**Pull request by kevinhowbrook**
:warning: based on https://github.com/torchbox/wagtail-torchbox/pull/561

### Description of Changes Made

This PR adds the fields needed for the course landing page and configures the subpage listing

### How to Test

1. Check out this branch
2. Add a course landing page
3. add some child course details pages
4. view the parent course landing page from step 2

Note: we don't have a requirement to configure a them for these pages as the designs are only dark, so `dark` has been hard coded as a page property.

Also note: not bothered with yaml as the pattern library is broken for pages

### MR Checklist

- [ ] Add a description of your pull request and instructions for the reviewer to verify your work.
- [ ] If your pull request is for a specific ticket, link to it in the description.
- [ ] Stay on point and keep it small so the merge request can be easily reviewed.
- [ ] Consider adding unit tests, especially for bug fixes. If you don't, tell us why.
- [ ] Tests and linting passes.
- [ ] Consider updating documentation. If you don't, tell us why.
- [ ] If relevant, list the environments / browsers in which you tested your changes.

### CODE REVIEW

File: tbx/courses/models.py [hunk 1]
```diff
@@ -27,6 +53,30 @@ class CourseLandingPage(utils_models.SocialFields, wagtail_models.Page):
...
+        index.SearchField("intro"),
+        index.SearchField("strapline"),
+    ]
+
+    @property
+    def theme(self):
+        # Don't offer a theme style, just set to dark
+        return "dark"
+
+    def get_subpages(self):
```
Comment (William-Blackie) [inline tbx/courses/models.py:66]: Suggestion: As this a private method (or atleast could be one), you could add an underscore to follow [PEP 0008](https://peps.python.org/pep-0008/#designing-for-inheritance)

File: tbx/courses/models.py [hunk 2]
```diff
@@ -27,6 +53,30 @@ class CourseLandingPage(utils_models.SocialFields, wagtail_models.Page):
         ),
     ]
 
+    search_fields = wagtail_models.Page.search_fields + [
+        index.SearchField("intro"),
+        index.SearchField("strapline"),
+    ]
+
+    @property
+    def theme(self):
```
Comment (William-Blackie) [inline tbx/courses/models.py:62]: Nitpick: As we're not computing a value, we don't need to define a method + decorator to do this, could just add it to model to do the same result, up to you.

File: tbx/project_styleguide/templates/patterns/molecules/course-grid/course-grid.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{% load wagtailcore_tags %}
+
+<div class="course-grid {% if not page.child_page_listing_heading %}course-grid--margin-top{% endif %}">
+    {% for card in cards %}
+        <div class="course-grid-item">
+            <p class="course-grid-item__sessions"><span>{{ card.sessions }}</span> {% if card.sessions and card.cost %} | {% endif %} {{ card.cost }}</p>
```
Comment (William-Blackie) [inline tbx/project_styleguide/templates/patterns/molecules/course-grid/course-grid.html:6]: Thought: Here we check if there is both `sessions & cost` but we assume `sessions` exists in the prior part, is this right?

File: tbx/project_styleguide/templates/patterns/pages/courses/course_detail_page.html [hunk 4]
```diff
@@ -5,23 +5,24 @@
...
-                <h2>
-                    {{ page.sessions }}
-                    {% if page.cost and page.sessions %} | {% endif %} {{ page.cost }}
-                </h2>
-            {% endif %}
+            <div class="title-block__course-detail">
+                {% if page.sessions %}
+                    <p class="title-block__sessions">
+                        {{ page.sessions }}
+                        {% if page.cost and page.sessions %} <span aria-hidden="true"></span> {% endif %} {{ page.cost }}
```
Comment (William-Blackie) [inline tbx/project_styleguide/templates/patterns/pages/courses/course_detail_page.html:15]: Thought: Similar question to the above point about `costs & sessions`

File: tbx/project_styleguide/templates/patterns/molecules/course-grid/course-grid.html [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{% load wagtailcore_tags %}
+
+<div class="course-grid {% if not page.child_page_listing_heading %}course-grid--margin-top{% endif %}">
+    {% for card in cards %}
+        <div class="course-grid-item">
```
Comment (William-Blackie) [inline tbx/project_styleguide/templates/patterns/molecules/course-grid/course-grid.html:5]: Suggestion: We have transitions on these cards, we might want to the whole card clickable for links / mobile users. 

I've done this with HTMX pretty easily by adding this line [here](https://github.com/wagtail/wagtail.org/pull/446/files#diff-8563055f2fe0b33d81a0d6ab21be1d35682347cf5a9d415b65b4fd03083a147fR5). Up to you if you want to go through bother of adding it.

**Review by William-Blackie — COMMENTED** [review]
Nice work Kevin :sparkles:, couple of nitpicks and suggestions.

File: tbx/project_styleguide/templates/patterns/pages/courses/course_detail_page.html (hunk 4 above)
Comment (kevinhowbrook) [reply tbx/project_styleguide/templates/patterns/pages/courses/course_detail_page.html]: fixedup to prev

**Pull request merged**